libs/coin-modules/coin-sui/src/logic/broadcast.test.ts

```diff
@@ -0,0 +1,77 @@
+import { broadcast } from "./broadcast";
+import suiAPI from "../network";
+
+// Mock the suiAPI
+jest.mock("../network", () => ({
+  __esModule: true,
+  default: {
+    executeTransactionBlock: jest.fn(),
+  },
+}));
+
+describe("broadcast", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  describe("with string input", () => {
+    it("should correctly extract transaction and signature and broadcast", async () => {
+      // Mock transaction string with 4-byte length prefix
+      // Example: "0010" (length 16 in hex) + "1234567890" (raw tx) + "abcdef" (signature)
+      const mockTransaction = "000a1234567890abcdef";
+
+      const mockDigest = "test-digest";
+      (suiAPI.executeTransactionBlock as jest.Mock).mockResolvedValue({ digest: mockDigest });
```

Comment:
using `jest.mocked` func would make the code readable https://jestjs.io/docs/upgrading-to-jest29#jestmocked